Packages/Core/Editor/Utils/GuiData.cs

```diff
@@ -0,0 +1,12 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityAtoms.Editor
+{
+	public class GuiData
```

Comment:
I just hated passing those 3/2 values into the methods because the methods were already having lots of variables to wrap your head around. So instead it was easier for me to just know that `GuiData` represents the parameters that pass from the `OnGUI`. Besides, I initially did have the `GuiData` as a struct but everything broke so I used a class and everything worked again as you would expect. So I just assumed that some of those fields change while the code is running so I had to use a class. Didn't look back on it since, but maybe I overlooked something I don't know

Reply:
Ok, now I see you mentioned the `in` keyword so maybe that's what I overlooked. Didn't know about this keyword and what it does so I'll research on it, thanks